src/classic/clvm_tools/stages/stage_2/module.rs

```diff
@@ -353,10 +354,11 @@ fn parse_mod_sexp(
                 constants.insert(name, quoted_decl);
                 Ok(())
             } else if op == "defconst".as_bytes() {
-                let r_of_declaration = rest(allocator, declaration_sexp)?;
-                let rr_of_declaration = rest(allocator, r_of_declaration)?;
-                let frr_of_declaration = first(allocator, rr_of_declaration)?;
-                delayed_constants.insert(name, frr_of_declaration);
+                // Use a new type-based match language.
+                let Rest::Here(Rest::Here(First::Here(definition))) =
+                    Rest::Here(Rest::Here(First::Here(ThisNode::Here))).
+                    select_nodes(allocator, declaration_sexp)?;
```

Comment:
it's not clear that this is any easier to read. but it's not a big deal.

Reply:
Was working up this match language for more general use in more complex optimization and compiling (and eventual refactoring) anyway.

Reply:
Resolving conversation (merge step)